sdk/dotnet/Pulumi/Serialization/Attributes.cs

```diff
@@ -94,4 +94,17 @@ public sealed class OutputConstructorAttribute : Attribute
     public sealed class EnumTypeAttribute : Attribute
     {
     }
+    
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ResourceIdentifierAttribute : Attribute
+    {
+        public string Token { get; }
```

Comment:
Nit: Should we name this `Type` to match the naming we've used elsewhere (same for the .ctor arg)?

https://github.com/pulumi/pulumi/blob/a5c30f1547c9865d55f6aaab4bb4c434d29b4202/sdk/dotnet/Pulumi/Resources/CustomResource.cs#L38